session/txn.go

```diff
@@ -292,7 +292,7 @@ func keyNeedToLock(k, v []byte, flags kv.KeyFlags) bool {
 	// do not lock row key for delete operation,
 	// lock primary key and unique index only.
 	if len(v) == 0 {
-		return flags.HasNeedLocked()
+		return flags.HasNeedLocked() || k[10] == 'r'
```

Comment:
It's better to use a const instead of a magic number here.

Reply:
There are a lot of magic number usage in TiDB. There may be another PR to group them together.

Reply:
You can use ` IsRowKey`.